File: smoke_tests/typescript/insecure_cookie.ts
// Insecure Cookie Test Cases

import { Request, Response } from 'express';

// Test 1: Cookie without Secure flag
function setInsecureCookie(res: Response): void {
    // VULNERABLE: Missing Secure flag - can be sent over HTTP
    res.cookie('session', 'abc123', { httpOnly: true });
}

// Test 2: Cookie without HttpOnly flag
function setJavaScriptAccessibleCookie(res: Response): void {
    // VULNERABLE: Missing HttpOnly - accessible via JavaScript
    res.cookie('authToken', 'xyz789', { secure: true });
}

// Test 3: Cookie without SameSite attribute
function setCookieWithoutSameSite(res: Response): void {
    // VULNERABLE: Missing SameSite - vulnerable to CSRF
    res.cookie('user', 'john', { secure: true, httpOnly: true });
}

// Test 4: Set-Cookie header directly without security flags
function setRawCookie(res: Response): void {
    // VULNERABLE: No security flags in raw Set-Cookie header
    res.setHeader('Set-Cookie', 'sessionId=secret123; Path=/');
}

// Test 5: Cookie with SameSite=None without Secure
function setSameSiteNoneCookie(res: Response): void {
    // VULNERABLE: SameSite=None requires Secure flag
    res.cookie('tracking', '12345', {
        sameSite: 'none',
        httpOnly: true
    });
}

File: smoke_tests/rust/insecure_cookie.rs
// Insecure Cookie Test Cases

use actix_web::{HttpResponse, cookie::Cookie};

// Test 1: Cookie without Secure flag
fn set_insecure_cookie() -> HttpResponse {
    // VULNERABLE: Missing Secure flag - can be sent over HTTP
    let cookie = Cookie::build("session", "abc123")
        .http_only(true)
        .finish();

    HttpResponse::Ok().cookie(cookie).finish()
}

// Test 2: Cookie without HttpOnly flag
fn set_javascript_accessible_cookie() -> HttpResponse {
    // VULNERABLE: Missing HttpOnly - accessible via JavaScript
    let cookie = Cookie::build("authToken", "xyz789")
        .secure(true)
        .finish();

    HttpResponse::Ok().cookie(cookie).finish()
}

// Test 3: Cookie without SameSite attribute
fn set_cookie_without_samesite() -> HttpResponse {
    // VULNERABLE: Missing SameSite - vulnerable to CSRF
    let cookie = Cookie::build("user", "john")
        .secure(true)
        .http_only(true)
        .finish();

    HttpResponse::Ok().cookie(cookie).finish()
}

// Test 4: Raw Set-Cookie header without security flags
fn set_raw_cookie() -> HttpResponse {
    // VULNERABLE: No security flags in raw Set-Cookie header
    HttpResponse::Ok()
        .append_header(("Set-Cookie", "sessionId=secret123; Path=/"))
        .finish()
}

// Test 5: Cookie with SameSite=None without Secure
fn set_samesite_none_cookie() -> HttpResponse {
    use actix_web::cookie::SameSite;

    // VULNERABLE: SameSite=None requires Secure flag
    let cookie = Cookie::build("tracking", "12345")
        .same_site(SameSite::None)
        .http_only(true)
        .finish();

    HttpResponse::Ok().cookie(cookie).finish()
}
